src/react/hooks/use-messages.ts

```diff
@@ -185,13 +223,57 @@ export const useMessages = (params?: UseMessagesParams): UseMessagesResponse =>
     ).unmount();
   }, [context, logger, onDiscontinuityRef]);
 
+  useEffect(() => {
+    if (!params?.reactionsListener) return;
+    return wrapRoomPromise(
+      context.room,
+      (room) => {
+        if (!params.reactionsListener) {
```

Comment:
Is this needed? This is checked above.

Reply:
I thought yes, because params might change before promise resolve. I saw that this isn't the case since the wrapper doesn't resolve anymore in that case.

But still, the check pleases the linter without a typecast so I'd leave it it.